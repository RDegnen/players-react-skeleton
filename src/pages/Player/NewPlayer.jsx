import React from 'react';
import { FormGroup, Label, Button, Input } from 'reactstrap';
import { withRouter } from 'react-router-dom';
import ReactRouterPropTypes from 'react-router-prop-types';

import constants from '../../constants';
import FormContainer from '../../components/FormContainer';

const createPlayer = (data, history) => {
  const authToken = localStorage.getItem('token');
  return fetch(`${constants.API_URL}/api/players`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${authToken}`,
    },
    body: JSON.stringify(data),
  }).then(res => res.json())
    .then(() => {
      history.push('/roster');
    })
    .catch(err => err);
};

const formGroups = [
  {
    field: 'first_name',
    type: 'text',
    id: 'firstName',
    label: 'First Name',
  },
  {
    field: 'last_name',
    type: 'text',
    id: 'lastName',
    label: 'Last Name',
  },
  {
    field: 'rating',
    type: 'number',
    id: 'rating',
    label: 'Rating',
  },
  {
    renderGroup: handleInput => (
      <FormGroup row>
        <Label for="handedness">Handedness</Label>
        <Input type="select" name="handedness" id="handedness" onChange={handleInput}>
          <option value="" disabled selected>Select a Hand</option>
          <option>left</option>
          <option>right</option>
        </Input>
      </FormGroup>
    ),
  },
  {
    renderGroup: () => (
      <FormGroup className="justify-content-center" row>
        <Button type="submit" value="Submit" id="create">Create</Button>
      </FormGroup>
    ),
  },
];

export const NewPlayer = ({ history }) => (
  <FormContainer history={history} formGroups={formGroups} submitAction={createPlayer} />
);

NewPlayer.propTypes = {
  history: ReactRouterPropTypes.history.isRequired,
};

export default withRouter(NewPlayer);
