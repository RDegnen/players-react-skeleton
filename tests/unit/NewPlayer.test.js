import React from 'react';
import { shallow } from 'enzyme';

import { NewPlayer } from '../../src/pages/Player/NewPlayer';

const getComponent = props => shallow(<NewPlayer history={{}} {...props} />);

describe('<NewPlayer />', () => {
  it('should match snapshot', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });
});
