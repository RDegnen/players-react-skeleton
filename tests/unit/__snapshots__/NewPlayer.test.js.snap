// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewPlayer /> should match snapshot 1`] = `
<FormContainer
  formGroups={
    Array [
      Object {
        "field": "first_name",
        "id": "firstName",
        "label": "First Name",
        "type": "text",
      },
      Object {
        "field": "last_name",
        "id": "lastName",
        "label": "Last Name",
        "type": "text",
      },
      Object {
        "field": "rating",
        "id": "rating",
        "label": "Rating",
        "type": "text",
      },
      Object {
        "renderGroup": [Function],
      },
      Object {
        "renderGroup": [Function],
      },
    ]
  }
  history={Object {}}
  submitAction={[Function]}
/>
`;
